src/ol_infrastructure/components/services/vault.py

```diff
@@ -394,28 +456,59 @@ def __init__(
                 organization=CERTIFICATE_CONFIG["organization"],
                 ou=CERTIFICATE_CONFIG["organizational_unit"],
                 postal_code=CERTIFICATE_CONFIG["zip_code"],
+                opts=ResourceOptions(parent=self.pki_intermediate_environment_backend),
             )
         )
 
-        # Sign genereated CSR for pki-intermediate-{env} backend by pki-intermediate-ca
+        # Create a default issuer role for pki-intermediate-{env}
+        self.pki_intermediate_environment_default_issuer_config = OLVaultPKIIntermediateRoleConfig(  # noqa: E501
+            pki_intermediate_backend=self.pki_intermediate_environment_backend,
+            role_name="default-issuer",
+            key_config=VaultPKIKeyTypeBits.rsa,
+            allowed_domains=["mit.edu"],
```

Comment:
We may also want to allow `.service.consul` domains, and optionally `.service.<env_name>.consul`

Reply:
I also think we might need to add another attribute if we were to allow subdomains - https://developer.hashicorp.com/vault/api-docs/secret/pki#allowed_domains

Reply:
Ooo Good catch! 

But, I kind of figured the default-issuer wouldn't really be utilized beyond this `bootstrapping` of the PKI itself? Seems to make more sense to create a new `role` or `issuer` for the use cases as the appear. But I defer to wiser minds if you guys disagree.